modules/lib/test/PageComponent.test-d.ts

```diff
@@ -0,0 +1,29 @@
+import type {
+    PageComponent,
+} from '../core/index';
+import {expectAssignable} from 'tsd';
+
+// On a page using automatic page template, the page component is empty.
+expectAssignable<PageComponent>({});
```

Comment:
One could argue that a page component without type: 'page' is a bug in xp. Cause it means when type is missing on a generic component, I have to assume it's a page.

Reply:
> One could argue that a page component without type: 'page' is a bug in xp

I was going to say something similar

If you put yourself in the application-developers shoes; is it useful to know that an empty object can be a `PageComponent`? 

If the app-developer writes a new component-object, do you want to enable it to be without `type`?

Reply:
> If you put yourself in the application-developers shoes; is it useful to know that an empty object can be a PageComponent?
No, but it's useful to know that a PageComponent can be an empty object :)

> If the app-developer writes a new component-object
This typically only happens in the context of testing.

> do you want to enable it to be without type?
I'm not too interested in control, I'm jut trying to represent reality.

Reply:
Seems to work